README.md

```diff
@@ -1,177 +1,248 @@
-```python
-study_name, exp_args_list = run_agents_on_benchmark(AGENT_4o, benchmark)
-study_dir = make_study_dir(RESULTS_DIR, study_name)
-run_experiments(n_jobs, exp_args_list, study_dir)
-```
+See [main.py](main.py) to launch experiments with a variety of options. This is like a lazy CLI that
+is actually more convenient. Just comment and uncomment the lines you need or modify at will (but
+don't push to the repo).
 
-use [main.py](main.py) to launch experiments with a variety
-of options. This is like a lazy CLI that is actually more convenient than a CLI.
-Just comment and uncomment the lines you need or modify at will (but don't push
-to the repo).
 
-<details>
+### Job Timeouts
 
-<summary>Debugging</summary>
+The complexity of the wild web, Playwright, and asyncio can sometimes cause jobs to hang. This
+disables workers until the study is terminated and relaunched. If you are running jobs sequentially
+or with a small number of workers, this could halt your entire study until you manually kill and
+relaunch it. In the Ray parallel backend, we've implemented a system to automatically terminate jobs
+exceeding a specified timeout. This feature is particularly useful when task hanging limits your
+experiments. 
 
-For debugging, run experiments using `n_jobs=1` and use VSCode debug mode. This
-will allow you to stop on breakpoints. To prevent the debugger from stopping
-on errors when running multiple experiments directly in VSCode, set
-`enable_debug = False` in `ExpArgs` 
-</details>
+### Debugging
 
+For debugging, run experiments with `n_jobs=1` and use VSCode's debug mode. This allows you to pause
+execution at breakpoints.
 
+### About Parallel Jobs
 
+Running one agent on one task corresponds to a single job. Conducting ablation studies or random
+searches across hundreds of tasks with multiple seeds can generate more than 10,000 jobs. Efficient
+parallel execution is therefore critical. Agents typically wait for responses from the LLM server or
+updates from the web server. As a result, you can run 10–50 jobs in parallel on a single computer,
+depending on available RAM.
 
+⚠️ **Note for (Visual)WebArena**: These benchmarks have task dependencies designed to minimize
+"corrupting" the instance between tasks. For example, an agent on task 323 could alter the instance
+state, making task 201 impossible. To address this, the Ray backend accounts for task dependencies,
+enabling some degree of parallelism. On WebArena, you can disable dependencies to increase
+parallelism, but this might reduce performance by 1–2%.
 
-<details>
+⚠️ **Instance Reset for (Visual)WebArena**: Before evaluating an agent, the instance is
+automatically reset, a process that takes about 5 minutes. When evaluating multiple agents, the
+`make_study` function returns a `SequentialStudies` object to ensure proper sequential evaluation of
+each agent. AgentLab currently does not support evaluations across multiple instances, but you could
```

Comment:
This is either too detailed or not detailed enough. What is a SequentialStudies object? What does it do?